soroban-env-host/src/vm.rs

```diff
@@ -75,6 +75,71 @@ pub struct VmFunction {
 }
 
 impl Vm {
+    fn check_contract_interface_version(
```

Comment:
Does this function gate deployment, or only execution, of a contract?

Reply:
both